common/web/types/src/ldml-keyboard/ldml-keyboard-xml-reader.ts

```diff
@@ -1,9 +1,16 @@
 import * as xml2js from 'xml2js';
-import { LDMLKeyboardXMLSourceFile } from './ldml-keyboard-xml.js';
+import { LDMLKeyboardXMLSourceFile, LKImport } from './ldml-keyboard-xml.js';
 import Ajv from 'ajv';
+import * as fs from 'fs';
 import { boxXmlArray } from '../util/util.js';
 
 export default class LDMLKeyboardXMLSourceFileReader {
+  readImportFile(version: string, subpath: string): Buffer {
+    // TODO-LDML: sanitize input string
+    let importPath = new URL(`../import/${version}/${subpath}`, import.meta.url);
+    return fs.readFileSync(importPath);
+  }
```

Comment:
File operations should be done through [`CompilerCallbacks`](https://github.com/keymanapp/keyman/blob/cb754353175f515e3677670ae1bf8776b7ce8f1c/developer/src/kmc/src/kmc.ts#L41-L57). This will allow us to do web-side compiler, as well as simplifying mocking for tests.

Reply:
done.